### PR TITLE
Document new argument (case_sensitive) for sync_partition_metadata

### PR DESCRIPTION
**Pull request by aalbu**

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -945,14 +945,20 @@ Procedures
...
     Check and update partitions list in metastore. There are three modes available:
 
     * ``ADD`` : add any partitions that exist on the file system, but not in the metastore.
     * ``DROP``: drop any partitions that exist in the metastore, but not on the file system.
     * ``FULL``: perform both ``ADD`` and ``DROP``.
 
+    The `case_sensitive` argument is optional and has a default value of `true`. For compatibility
+    with Hive's `MSCK REPAIR TABLE`, the partition names on the file system (the directory name
+    part corresponding to the column name) are expected to be in lowercase (e.g. `col_x=SomeValue`).
+    If that is not the case, those paths will be ignored, unless a value of `false` is passed for
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/hive.rst:959]: The ``case_sensitive`` argument is optional. The default value is ``true`` and compatible with Hive's ``MSCK REPAIR TABLE``. Partition names, and therefore the directory names on the file system, need to use lowercase such as ``col_x=my-value``.  Mixed case partitions are ignored, unless the argument is set to ``false``.

Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive.rst:959]: the above should fix grammar, clarity and source formatting issues. I have not test it though..

Comment (aalbu) [reply presto-docs/src/main/sphinx/connector/hive.rst:959]: Thanks for fixing the formatting. I picked `col_x=SomeValue` deliberately as an example, to make it clear that the part corresponding to the column name is what needs to be lower case in Hive compatibility mode.  The actual partition value can have any capitalization.

I am open to suggestions, but this is neither accurate nor clear:
> Partition names, and therefore the directory names on the file system, need to use lowercase

**Comment by findepi** [comment]
```
[ERROR] Failed to execute goal io.airlift.maven.plugins:sphinx-maven-plugin:2.1:generate (default) on project presto-docs:
Failed to run the report: Sphinx report generation failed -> [Help 1]
```

File: presto-docs/src/main/sphinx/connector/hive.rst (hunk 1 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive.rst:959]: Feel free to merge the two as desired ... but you need to fix the syntax to ensure this builds.

**Pull request merged**